src/Button/index.js

```diff
@@ -2,60 +2,112 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
 
-import { Container, ContainerIconLeft, ContainerIconRight } from './elements';
+import { Container, ContainerIcon } from './elements';
 
-const displayIconLeft = (side, icon) => {
-  return side === 'right' || icon === null ? null : <ContainerIconLeft>{icon}</ContainerIconLeft>;
-};
-const displayIconRight = (side, icon) => {
-  return side === 'left' || icon === null ? null : <ContainerIconRight>{icon}</ContainerIconRight>;
-};
+/**
+ * Button
+ *
+ * This component is in charge of displaying
+ * a button
+ *
+ * @param {string} appearance // The base styling to apply to the button.
+ * @param {node} children // Anything that can be rendered: numbers, strings, elements or an array (or fragment).
+ * @param {string} className // ClassName needed by styled components.
+ * @param {bool} fluid // Whether the button is fluid or not.
+ * @param {bool} disabled // Whether the button is disabled or not.
+ * @param {node} icon // Add icon node here to illustrate the text aside.
+ * @param {enum} icon // Set the icon position compared to the text. It will be set to the left side by default.
+ * @param {function} onClick // Handler of click on the button.
+ * @param {enum} size // Whether the button size is small, default or large. It will be set to default by default.
+ * @param {enum} type // Whether it is a button or a form submission.
+ *
+ * @return {jsx}
+ */
 
-const Button = ({ children, icon, iconPosition, ...rest }) => {
+const Button = ({
+  appearance,
+  children,
+  className,
+  disabled,
+  fluid,
+  icon,
+  iconPosition,
+  onClick,
+  size,
+  type,
+  ...rest
+}) => {
   return (
-    <Container {...rest}>
-      {displayIconLeft(iconPosition, icon)} {children} {displayIconRight(iconPosition, icon)}
+    <Container
+      appearance={appearance}
+      className={className}
+      disabled={disabled}
+      fluid={fluid}
+      icon={icon}
+      iconPosition={iconPosition}
+      onClick={onClick}
+      size={size}
+      type={type}
+      {...rest}
+    >
+      {displayIcon(iconPosition, icon)}
+      {children}
     </Container>
   );
 };
 
-const { node, func, string, bool } = PropTypes;
+/**
+ * displayIcon
+ *
+ * This function is a helper to place
+ * the icon correctly inside a button.
+ * He return an react element with the
+ * good props: left or right.
+ *
+ * @param {string} side // The side where the icon should be positioned [left or right].
+ * @param {jsx} icon // The icon elements.
+ *
+ * @return {jsx}
+ */
+const displayIcon = (side, icon) => {
+  if (!side || !icon) return null;
+  return {
+    left: <ContainerIcon left>{icon}</ContainerIcon>,
+    right: <ContainerIcon right>{icon}</ContainerIcon>,
+  }[side];
+};
+
+/**
+ * PropTypes Validation.
+ */
+const { bool, func, node, oneOf, string } = PropTypes;
 Button.propTypes = {
+  appearance: oneOf(['default', 'primary', 'secondary', 'success', 'danger', 'google']),
```

Comment:
Using enum is a good thing, but what happened if we pass something wrong as `appareance` ? I think this will break code

Reply:
@AntoninMarchardDev That's why propType exist dude !

Reply:
@spartDev I agree. But imagine I pass something like `priary` with an error in typo. Does proptypes will put `default` by default ?

Reply:
No But react send you an error

Reply:
Is that a good idea to have an error when you use a component ? I take place of a lambda user, that will download your package, I prefer to see a default button if I made a typo mistake instead of a blank page with an error, isn't it ?

Reply:
@AntoninMarchardDev But Its the Goal of the propType to catch error. Otherwise what's the point?

Reply:
@AntoninMarchardDev I disagree. This is why you have development environment. I think this shouldn't be changed.
As a user, I prefer having an error telling me I did a typo than having a silent "error".

Reply:
@leopoldhoudin @spartDev ok I agree. I was just imagine If I'm using something like "bulma" or "bootstrap", I want to have a default button if I made a typo mistake, and not a blank page. But If you think it's better to have a big error for a typo, it's ok for me